### ISSUE TITLE
SMB connection not possible

### ISSUE THREAD

**Issue by Sagitee** [celzero/rethink-app#1159]
Hi, I have recently discovered Rethink and I was positively surprised to find that it surpasses the powerful NetGuard, that I was using previously, in many aspects. Thank you for this nice and very effective app.
I would like to mention the following issue.
I have setup a MixPlorer file manager SMB server on each of two Android 13 LOS mobiles, that had been working fine while I used NetGuard.
Rethink seems to block the server connection of MixPlorer (without showing any relative blocking in the logs) and it allows the connection only when I set MixPlorer as excluded in the FW rules.
In the logs are shown only the client side connections, which are set as allowed and are properly allowed by Rethink.

Thank you.

**Comment by ThreeDeeJay** [comment]
I'm not sure the extent of unwanted consequences since I think it'd allow LAN access for *all* apps, but you could try enabling Configure > Network > Do not route Private IPs (experimental). IIRC I had to use a similar option with NetGuard to access SMB servers with MiXplorer. If that still doesn't work, you might have to whitelist the server IP for MiXplorer, just in case.
I just wish we could allow LAN access on a per-app basis like with AFWall+, but maybe this is a limitation of using a VPN/non-root firewall.

**Comment by Sagitee** [comment]
Thank you for your response.

Your suggestion works but as you mentioned it allows unconditional (no logs, no rules application etc.). LAN access for all apps, which i think is less secure than setting MixPlorer as excluded temporarily in the Rethink's FW settings.

> maybe this is a limitation of using a VPN/non-root firewall

As I said it works fine with the non-root NetGuard (without one having to enable the (settings/network options/subnet routing/allow LAN access) option that you mentioned).

This FAQ
https://github.com/M66B/NetGuard/blob/master/FAQ.md#user-content-faq45
says:
The Android VPN service handles outgoing connections only (from applications to the internet), so incoming connections are normally left alone.

If this is valid for Rethink too, it seems that in some way Rethink can intervene (not expected) in that traffic, as isolate or exclude FW modes can block or allow it, but it cannot apply any defined IP whitelisting rule (something expected as it's about an ingoing connection).

**Comment by ignoramous** [comment]
> I just wish we could allow LAN access on a per-app basis like with AFWall+, but maybe this is a limitation of using a VPN/non-root firewall.

You're correct. This isn't possible to do with the current Android VPN APIs.

> Your suggestion works but as you mentioned it allows unconditional (no logs, no rules application etc.). LAN access for all apps, which i think is less secure than setting MixPlorer as excluded temporarily in the Rethink's FW settings.

Most apps don't usually initiate any connection to private IPs, so you're good either way (*excluding* MixPlorer or enabling *Do not route Private IPs*).

That said, the next version (`v055b`) has support for Multicast DNS (#1005), which some of the P2P / LAN apps like Syncthing / FritzBox (and MixPlorer?) require to discover services and devices on LAN. May be, if it works like I expect it to, you'd not have to exclude or unroute Private IPs.

**Comment by yuan8886** [comment]
如果我将电脑A设置为SMB服务器，我可以使用手机B或者手机C上的mixplorer访问电脑上的文件。但是，如果我使用mixplorer将手机B设置为SMB服务器，电脑A缺无法访问手机上面的文件。可以打开对应的地址链接，但是不会显示任何文件内容，界面是空白的，打开时系统也没有提示我输入账号密码。手机C是可以访问手机B的SMB服务器文件的。如果使用手机C创建服务器，手机B也可以访问，但电脑A仍旧无法访问。我以前尝试过使用其他app在手机上创建smb服务器，是可以被电脑访问的，但是速度太慢了，并且经常掉线，所以我想尝试使用mixpolrer

**Comment by yuan8886** [comment]
If I set computer A as an SMB server, I can use mixplorer on phone B or phone C to access the files on the computer. However, if I use mixplorer to set phone B as an SMB server, computer A cannot access the files on the phone. I can open the corresponding address link, but it does not display any file content, and the interface is blank. The system does not prompt me to enter an account and password when I open it. Phone C can access the SMB server files on phone B. If I use phone C to create the server, phone B can also access it, but computer A still cannot access it. I have previously tried using other apps to create an SMB server on the phone, which can be accessed by the computer, but the speed is too slow, and it often drops out, so I want to try using mixplorer

**Comment by Sagitee** [comment]
@yuan8886 
Access to a Rethink firewalled phone (from either a pc or a phone) is only possible  in the cases No 3 and 4, as described in the issue #2289.

**Comment by Sagitee** [comment]
I'm closing this old issue as I think that issue #2289 describes the same problem in a more accurate and comprehensive way. It is happening to any app and with any type of inbound connection (HTTP, FTP etc). So it is not a MixPlorer or an SMB only issue.

**Comment by ignoramous** [comment]
Hm, *some* apps may become LAN discoverable if *Configure -> Network -> Endpoint Independent mapping* is turned ON. If you have already tried, did that setting not work for you?

**Comment by Sagitee** [comment]
Thank you for replying.
I just tried it, but it doesn't work.
I'm on 055t on both phones.
